tables/Tables/TableUtil.cc

```diff
@@ -0,0 +1,321 @@
+      if (tableName.empty()) {
+        throw TableError
+          ("Empty string provided for tableName; will not attempt delete.");
+      }
+      // See if the name represents a table.
+      if (! Table::isReadable(tableName)) {
+        // See if the name contains subtable names using ::
+        std::pair<Table,String> t = tryFindSubTable (tableName, False);
+        if (! (t.first.isNull()  ||  t.second.empty())) {
+          deleteSubTable (t.first, t.second, checkSubTables);
+          return;
+        }
+      }
+      // Delete the table (which fails if it is not a table or still in use).
+      String tabName = Path(tableName).absoluteName();
+      String message;
+      if (! canDeleteTable (message, tabName, checkSubTables)) {
+        throw (TableError ("Table " + tabName + " cannot be deleted: " +
+                           message));
+      }
+      Table table(tabName, Table::Delete);
+    }
+
+
+    void deleteSubTable (Table& parent, const String& subtableName, Bool checkSubTables)
+    {
+      String message;
+      if (! canDeleteSubTable (message, parent, subtableName, checkSubTables)) {
+        throw (TableError ("Subtable " + subtableName + " in " +
+                           parent.tableName() + " cannot be deleted: " +
+                           message));
+      }
+      Table subtab = parent.keywordSet().asTable(subtableName);
+      subtab.markForDelete();
+      // If there, remove the keyword referring the subtable.
+      Int inx = parent.keywordSet().fieldNumber(subtableName);
+      if (inx >= 0) {
+        if (parent.keywordSet().type(inx) == TpTable) {
+          parent.reopenRW();
+          parent.rwKeywordSet().removeField (subtableName);
+        }
+      }
+    }
+
+    //# The logic is similar to that in Table::open.
+    rownr_t getLayout (TableDesc& desc, const String& tableName)
+    {
+      rownr_t nrow;
+      uInt format;
+      String tp;
+      AipsIO ios (Table::fileName(getFullName(tableName)));
+      uInt version = ios.getstart ("Table");
+      if (version > 3) {
+        throw TableError ("Table version " + String::toString(version) +
+                          " not supported by TableUtil in this version of Casacore");
+      }
+      if (version > 2) {
+        ios >> nrow;
+      } else {
+        uInt n;
+        ios >> n;
+        nrow = n;
+      }
+      ios >> format;
+      ios >> tp;
+      if (tp == "PlainTable") {
+	PlainTable::getLayout (desc, ios);
+      } else if (tp == "RefTable") {
+        RefTable::getLayout (desc, ios);
+      } else if (tp == "ConcatTable") {
+        ConcatTable::getLayout (desc, ios);
+      } else {
+        throw (TableInternalError
+               ("TableUtil::getLayout: unknown table kind " + tp));
+      }
+      ios.close();
+      return nrow;
+    }
+
+    TableInfo tableInfo (const String& tableName)
+      {
+        return BaseTable::tableInfo (getFullName(tableName));
+      }
```

Comment:
I've replaced the tabs by spaces and re-indented all code.
An automatic tool would generate a massive change. Furthermore, different casacore developers usually like their own style and I think it would be hard to find a compromise.